components/configuration-mgt/org.wso2.carbon.identity.configuration.mgt.core/src/main/java/org/wso2/carbon/identity/configuration/mgt/core/ConfigurationManager.java

```diff
@@ -119,6 +119,20 @@ Resources getTenantResources(String tenantDomain, Condition searchCondition) thr
      */
     Resource addResource(String resourceTypeName, ResourceAdd resourceAdd) throws ConfigurationManagementException;
 
+    /**
+     * This API is used to create the given resource including a file.
+     *
+     * @param resourceTypeName Name of the {@link ResourceType}.
+     * @param resourceAdd      Request to create the {@link Resource}.
+     * @param resourceAdd      Name of the {@link ResourceFile}
+     * @param fileName         {@link InputStream} representing the file.
+     * @return 201 created. Returns {@link Resource} created.
+     * @throws ConfigurationManagementException Resource management exception.
+     */
+    Resource addResourceWithFile(String resourceTypeName, ResourceAdd resourceAdd, String fileName,
```

Comment:
Do we need to add default implementation for the newly introduced methods to this interface.

Reply:
Addressed in 3d05632 and 7d912de